### DISCUSSION TITLE
Executing `phcode` in terminal does nothing.

### DISCUSSION THREAD

**Discussion by anoduck** [phcode-dev/phoenix#1832] [Q&A]
It might have to do with running Kali Linux and not your standard Debian or Ubuntu, but when I attempt to run phoenix from the terminal with `phcode`, it does nothing and never backgrounds or outputs any errors. I literally type `phcode`, then hit `<enter>`, the cursor drops to the next line, and it stays that way until I hit `Ctrl-c` several minutes later.

**Comment by charlypa** [comment]
@anoduck 
are you able to launch application from start menu?

**Reply by anoduck** [reply]
@charlypa Running wayland with sway, so the start menu is jgmenu. It appears, executing phcode from the start menu has the exact same effect as running it from the terminal. Phcode starts a single process, but then immediately sleeps. 

I can run `~/.phoenix-code/phnode src-node/index.js`, but get `not found` when attempting to visit the urls from a web browser. 

```bash
$ ../phnode index.js
Server Opened on port:  45711
Server running on http://localhost:45711
Phoenix node Static Server url is http://localhost:45711/StaticbOCJKjzd
Phoenix node tauri FS url is ws://localhost:45711/PhoenixFSrWSTuWtU
Phoenix node connector url is ws://localhost:45711/PhoenixNodeRPX4iaM1
Phoenix live preview comm url is ws://localhost:45711/PreviewCommzQL1r9fL
System Node.js (22.8.0) is newer than phnode(20.15.0). Using system Node.js.
```

----

`~/.phoenix-code/phoenix-code: ELF 64-bit LSB pie executable, x86-64, version 1 (SYSV), dynamically linked, interpreter /lib64/ld-linux-x86-64.so.2, BuildID[sha1]=83c6430872c5d46da26b64b0c8b275e90e2b7cee, for GNU/Linux 3.2.0, not stripped`

**Reply by charlypa** [reply]
@anoduck 
Which desktop are you using kde or gnome?

**Reply by charlypa** [reply]
@anoduck 
i just tried in kali linux it works for me I use xfce and in virtual box 
![image](https://github.com/user-attachments/assets/852a7b30-7b74-47c0-9cfa-bdc418f2ef16)

**Reply by anoduck** [reply]
@charlypa This explains a few things, and it is good to see it can work on Gnome or KDE. Currently, running Sway with Kali, but this does tell us there might be a requirement or two not satisfied as a result of my desktop selection. 

Does phcode generate any logging output that I can inspect? Is there a way to place phcode into debug mode? I am not experienced enough with nodejs to have figured this one out.

**Reply by charlypa** [reply]
@anoduck 
Technically we just need webkit alone to work desktop type does not matter. There are a few cases where graphics drivers could cause issues.  

@abose would you guide @anoduck on how to debug?

**Reply by anoduck** [reply]
@charlypa It might be the case my old development box is just too old. It is a real possibility.

**Reply by abose** [reply]
phnode is not supported to be executed directly, so the logs above for phnode can be ignored.

Try running phoenix-code binary directly and you should see any errors if any in console.

Please try this docs https://docs.phcode.dev/docs/Linux/#manual-installation

**Reply by anoduck** [reply]
Exactly the same result with no output. The cursor drops to the next line, there is a short increase in system load, and then the program sleeps. I have even gone so far as building Phoenix manually, and it didn't make a difference.

~~Since the configuration of Sway on Kali is something wholly of my own concoction, the ability to reproduce the error is questionable under best circumstances. Kali Linux does have an official port of Kali for the i3 Desktop, something I have wanted to revert to. This should provide nearly identical testing conditions to my current Desktop, less one critical factor, me.~~

~~So, unless similar issues occur with Sway involving different users, I suggest we put this one on the back burner, and I will revisit when I reconfigure my desktop to run I3.~~

**Reply by anoduck** [reply]
Reopening this issue since it was never resolved. The desktop has migrated from sway to hyprland, and nothing has changed. I was wondering if more headway has been made towards discovering what could possibly be causing this issue.
